bytecomp/bytepackager.ml

```diff
@@ -53,44 +65,53 @@ let empty_state = {
   debug_dirs = String.Set.empty;
   primitives = [];
   offset = 0;
-  mapping = Ident.Map.empty;
+  mapping = GlobalMap.empty;
   subst = Subst.identity;
 }
 
 (* Update a relocation.  adjust its offset, and rename GETGLOBAL and
    SETGLOBAL relocations that correspond to one of the units being
    consolidated. *)
 
+let ident_of_global (Global id) = Ident.create_persistent id
+
 let rename_relocation packagename objfile mapping base (rel, ofs) =
   let rel' =
     match rel with
-      Reloc_getglobal id ->
+      | Reloc_getglobal {predef = true; _} -> rel
+      | Reloc_getglobal {predef = false; id} ->
```

Comment:
Originally the suggestion to have a `predef : bool` parameter for `Reloc_getglobal` was because the two cases `getpredef` and `getglobal` would do essentially the same thing. Now, if I understand correctly, `Reloc_getglobal {predef = true; id}` relocations are completely ignored? What is the point of keeping them around in this case?

I'm not sure what is the intention with "predef globals". Do they make any sense? Were they a weird bug before that you have removed from the codebase? Completely? Only in some places?